libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/statistics/KotlinBuildStatsService.kt

```diff
@@ -106,21 +105,26 @@ internal abstract class KotlinBuildStatsService internal constructor() : BuildAd
                             )
                             instance = JMXKotlinBuildStatsService(mbs, beanName)
                         } else {
+                            val kotlinBuildStatProvider = project.provider{ KotlinBuildStatListener(beanName) }
                             val newInstance = DefaultKotlinBuildStatsService(gradle, beanName)
+
+                            if (isConfigurationCacheAvailable(gradle)) {
+                                listenerRegistryHolder.listenerRegistry!!.onTaskCompletion(kotlinBuildStatProvider)
```

Comment:
:x: this won't work with the configuration cache, the listener passed to `onTaskCompletion()` must be a registered build service as explained at https://docs.gradle.org/current/userguide/build_services.html#receiving_information_about_task_execution, not an arbitrary provider

Reply:
Yes, stats isn't supported yet. fix for is is move to a separate issue

Reply:
Got it. But, then why register this at all? It won't work